Update ModernCampus Lumens.ps1

T4E-JAnderson opened the pull request:
Add writeback methods
Various Bugixes

T4E-JAnderson committed:
Update ModernCampus Lumens.ps1

Add writeback functions
Various Bugixes

msheldont4e commented on ModernCampus Lumens.ps1:
```diff
@@ -593,3 +1292,210 @@ function New-ModernCampusLumensConnection {
+                                        <CreateDateEnd><![CDATA[2078-01-01]]></CreateDateEnd>
+                                    </Criteria>
+                                </Request>
+                            </Wrapper>' -f $i, ($i+$system_params.pagesize)  # <NewOrModifiedSinceLastTransfer><![CDATA[Yes]]></NewOrModifiedSinceLastTransfer>
```
Remove commented out code

msheldont4e commented on ModernCampus Lumens.ps1:
```diff
@@ -593,3 +1292,210 @@ function New-ModernCampusLumensConnection {
+    }
+}
+
+# (([xml]$response).Wrapper.ResponseData.ClassDetails | ? {$_.Classes.TotalSeats -ne $_.Classes.RemainingSeats})[10].classes.RosterDetails.Learner[0]
```
Remove commented out code

msheldont4e commented on ModernCampus Lumens.ps1:
```diff
@@ -593,3 +1292,210 @@ function New-ModernCampusLumensConnection {
+                                        <CreateDateEnd><![CDATA[2078-01-01]]></CreateDateEnd>
+                                    </Criteria>
+                                </Request>
+                            </Wrapper>' -f $i, ($i+$system_params.pagesize)  # <NewOrModifiedSinceLastTransfer><![CDATA[Yes]]></NewOrModifiedSinceLastTransfer>
```
Remove Commented out code.

msheldont4e commented on ModernCampus Lumens.ps1:
```diff
@@ -420,75 +664,520 @@ function Idm-InstructorsRead {
+            semantics = 'create'
+            parameters = @( 
+                @{ name = 'LearnerID';         allowance = 'mandatory'  },
+                @{ name = 'FieldName';allowance = 'mandatory'   },
```
Future consideration is to use the options for the global:properties, we do this with other connectors

T4E-JAnderson replied:
This was mostly a copy of the Instructor CustomData set call, though the required fields were different.  Thus, it wasn't rewritten to use the Global:Properties.

msheldont4e commented on ModernCampus Lumens.ps1:
```diff
@@ -420,75 +664,520 @@ function Idm-InstructorsRead {
+                @{ name = 'FirstName';allowance = 'mandatory' }
+                @{ name = 'LastName';allowance = 'mandatory' }
+                @{ name = 'UserName';allowance = 'mandatory' }
+                @{ name = 'FieldName';allowance = 'mandatory'   },
```
Future consideration, use global:properties options

T4E-JAnderson replied:
The challenge here using the global:properties is the Set Custom Data call requires fields that are not part of the CustomData Get call.

T4E-JAnderson committed:
Update ModernCampus Lumens.ps1

Remove old commented out code.

T4E-JAnderson committed:
Update ModernCampus Lumens.ps1

Fix Class CustomDetails pull.

T4E-JAnderson committed:
Update ModernCampus Lumens.ps1

Lumens changed API.  Added ResponseData wrapper back to ClassDetails and LumensLearners response data.
